src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -199,5 +199,38 @@ public static void Insert_CharSpan_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("index", () => builder.Insert(builder.Length + 1, new ReadOnlySpan<char>(new char[0]))); // Index > builder.Length
             AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Insert(builder.Length, new ReadOnlySpan<char>(new char[1]))); // New length > builder.MaxCapacity
         }
+
+        public static IEnumerable<object[]> Equals_String_TestData()
+        {
+            var mediumString = new string('a', 30);
+            var largeString = new string('a', 1000);
+            var extraLargeString = new string('a', 41000); // 8000 is the maximum chunk size
+
+            var sb1 = new StringBuilder("Hello");
+            var sb2 = new StringBuilder(20).Append(mediumString);
+            var sb3 = new StringBuilder(20).Append(largeString);
+            var sb4 = new StringBuilder(20).Append(extraLargeString);
+
+            yield return new object[] { sb1, "Hello", true };
+            yield return new object[] { sb1, "Hel", false };
+            yield return new object[] { sb1, "Hellz", false };
+            yield return new object[] { sb1, "Hellz0", false };
+            yield return new object[] { sb1, "", false };
+            yield return new object[] { new StringBuilder(), "", true };
+            yield return new object[] { sb2, mediumString, true };
+            yield return new object[] { sb2, "H", false };
```

Comment:
Is it interesting to add test cases `(sb2, largeString, false)` or `(sb3/sb4, mediumString/largeString, false)`? (checking substring comparison down the road)

Maybe even difference further out in the string - `mediumString/largeString+"b"`? (I didn't check the implementation, just guessing based on what I heard about it)

Reply:
The first check is a length check in the implementation and these will just only result in length mismatch and return false.

Reply:
I see, in that case it might be interesting to add 1 case with large string, differing in the last character.